examples/moving_least_squares/mls.hpp

```diff
@@ -0,0 +1,117 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#pragma once
+
+#include <ArborX.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <cassert>
+
+#include "mls_computation.hpp"
+#include "mpi_comms.hpp"
+
+template <typename MemorySpace, typename Points>
+struct TargetPoints
+{
+  Points target_points;
+  std::size_t num_neighbors;
+};
+
+template <typename MemorySpace, typename Points>
+struct ArborX::AccessTraits<TargetPoints<MemorySpace, Points>,
+                            ArborX::PredicatesTag>
+{
+  static KOKKOS_FUNCTION std::size_t
+  size(TargetPoints<MemorySpace, Points> const &tp)
+  {
+    return tp.target_points.extent(0);
+  }
+
+  static KOKKOS_FUNCTION auto get(TargetPoints<MemorySpace, Points> const &tp,
+                                  std::size_t i)
+  {
+    return ArborX::nearest(tp.target_points(i), tp.num_neighbors);
+  }
+
+  using memory_space = MemorySpace;
+};
+
+template <typename ValueType, typename PolynomialBasis, typename RBF,
+          typename MemorySpace>
```

Comment:
`ValueType` (and `FloatingPoint` in your example) is the type of the calculations and the stored matrices, but it might be different from the coordinate type and the evaluation type. `FloatingCalculationType` would be a better name?

Reply:
> `FloatingCalculationType` would be a better name?

May be a bit better than a generic `ValueType`, but it still does not quite express the intent. There must be a better name.